app/code/Magento/Catalog/Model/Product/Image/Cache.php

```diff
@@ -127,4 +138,25 @@ protected function processImageData(Product $product, array $imageData, $file)
 
         return $this;
     }
+
+    /**
+     * Retrieve themes which are currently active in the frontend
+     *
+     * @return array
+     */
+    protected function getThemesInUse()
+    {
+        $themesInUse = [];
+
+        $registeredThemes = $this->themeCollection->loadRegisteredThemes();
+        $storesByThemes   = $this->themeCustomizationConfig->getStoresByThemes();
```

Comment:
Themes can be set not only on store level, but on product/category/cms. Please addthe logic which would include these cases as well.

Reply:
@vrann @hostep I've used this code and added themes checking for produt, category and pages.
https://github.com/denisristic/magento2/commit/8308ae1ef89ace1f5a56a8aee73812f9290e80aa#comments

Reply:
@vrann Is it actually necessary to check for product/category/page?
Even if any of them have assigned custom theme that is disabled it would not matter because it would not be displayed because theme is disabled?